### PR TITLE
qt: Set C locale for amountWidget

### PR DESCRIPTION
**Pull request by hebasto**
Fix #13873

### CODE REVIEW

File: src/qt/transactionview.h [hunk 1]
```diff
@@ -70,6 +70,7 @@ class TransactionView : public QWidget
     QComboBox *watchOnlyWidget;
     QLineEdit *search_widget;
     QLineEdit *amountWidget;
+    QDoubleValidator *amountValidator;
```
Comment (luke-jr) [inline src/qt/transactionview.h]: This isn't used anywhere. Probably we either need to delete it in cleanup, or don't need to store it at all.

**Review by luke-jr — CHANGES_REQUESTED** [review]
Not sure if this is a good idea. What is the practical effect of setting the locale here?

Can a correct/point-based number still be entered and work properly?

**Comment by laanwj** [comment]
we don't use the system locale for bitcoin amount entry, so this seems right

utACK bd5107055d393e14269faca84741c8167500e7b5

File: src/qt/transactionview.h (hunk 1 above)
Comment (laanwj) [reply src/qt/transactionview.h]: it is used: it's used to validate what is entered in the amount field

Comment (laanwj) [reply src/qt/transactionview.h]: Oh I agree with @luke-jr actually; there's no need to store this in the object, just store it temporarily in a local variable in `TransactionView::TransactionView` while you need to manipulate it.

File: src/qt/transactionview.cpp [hunk 2]
```diff
@@ -106,7 +106,9 @@ TransactionView::TransactionView(const PlatformStyle *platformStyle, QWidget *pa
     } else {
         amountWidget->setFixedWidth(100);
     }
-    amountWidget->setValidator(new QDoubleValidator(0, 1e20, 8, this));
+    amountValidator = new QDoubleValidator(0, 1e20, 8, this);
```
Comment (laanwj) [inline src/qt/transactionview.cpp]: e.g. do
```c++
QDoubleValidator *amountValidator = new QDoubleValidator(0, 1e20, 8, this);
```

**Comment by hebasto** [comment]
@luke-jr @laanwj 
Thank you for your reviews.
Fixed.

**Comment by hebasto** [comment]
I just found out that there is another PR addressed this issue: #13278
@MarcoFalke the @DrahtBot did not report about a PR conflict. Is it ok?

**Comment by MarcoFalke** [comment]
@hebasto The bot can't check against PRs that conflict with master.

**Comment by jonasschnelli** [comment]
Unsure.
Current master won't let me enter a comma (`,`)... it forces me to use "." for the dec. separator.
While in this PR, I can enter a comma (gets ignored) which may confuse users.
Thoughts?

**Comment by hebasto** [comment]
@jonasschnelli 
Thank you for your review.

> While in this PR, I can enter a comma (gets ignored) which may confuse users.

The comma `,` is a group separator in the C locale.

> Thoughts?

Fixed. Please re-review.

**Comment by Sjors** [comment]
On a macOS machine with Dutch locale (`,` decimal separator) When I enter an amount with a comma in the Send screen it's automatically converted to a dot, which is nice.

When I enter a comma in the search screen it's ignored, as @jonasschnelli noticed. That's annoying, but at least it fixes the bug. The user will see a list of amounts with `.` separators so they'll probably figure it out.

tACK b0510d7

**Pull request merged**